test/events_test.exs

```diff
@@ -1,3 +1,120 @@
 defmodule EventsTest do
   use ExUnit.Case
+
+  defmodule TestSubscriber do
+    def echo_event(event, value, %{send_to: pid} = config) do
+      send(pid, {:event, event, value, config})
+    end
+
+    def raise_on_event(_event, _value, _config) do
+      raise "Got an event"
+    end
+  end
+
+  setup do
+    subscription_id = :crypto.strong_rand_bytes(16) |> Base.encode16()
+
+    on_exit fn ->
+      Events.unsubscribe(subscription_id)
+    end
+
+    {:ok, sub_id: subscription_id}
+  end
+
+  test "subscribing returns error if subscription with the same ID already exist", %{
+    sub_id: sub_id
+  } do
+    :ok = Events.subscribe(sub_id, [:some, :event], TestSubscriber, :echo_event)
+
+    assert :error = Events.subscribe(sub_id, [:some, :event], TestSubscriber, :echo_event)
```

Comment:
Would returning a more descriptive error (something like `{:error, :exists}`) make more sense?

Reply:
Yes, it definitely would! There should probably be a counterpart for `unsubscribe/1`, too.